sarracenia/__init__.py

```diff
@@ -748,7 +748,10 @@ def fromFileInfo(path, o, lstat=None):
         # rename path given with no filename
 
         if o.rename:
-            msg['retrievePath'] = msg['new_retrievePath']
+            # ensure we have the old name for retrieval usage.
+            if 'new_retrievePath' in msg:
+                msg['retrievePath'] = msg['new_retrievePath']
+
```

Comment:
I'm still confused about why the retrievePath code is inside `if o.rename`? I might be overlooking something, but I think the two things are independent? Rename is telling the subscriber what name to give the file when it downloads it, retrievePath tells the subscriber where to download from. But retrievePath should work whether or not rename is set?

Reply:
I don't understand it either... I didn't think that far.  I think I am confused about how rename works... relation between *relPath* and *rename* fields... e.g. should rename overwrite *relPath*? in which case, the actual path you use to download should be *retrievePath* .  ... eg.. one way to implement rename... is to copy relPath to retrievePath, and then apply the rename to the relPath and download normally... I don't recall if that's how things ended up or not...

Reply:
I think the implementation vacilated back and forth between changing *relPath* and using the *rename* field... so there are leftovers of each... and can switch between the two easily.

Advantage of *relPath* approach is can specify entire directory tree at the same time...with *rename* it is only the file name.

Reply:
so I think it is correct as-is... for some use cases... but would be a lot more work to explore thoroughly...